common/spdat.cpp

```diff
@@ -1086,437 +1371,551 @@ bool IsRegularSingleTargetHealSpell(uint16 spell_id)
+		case SE_LimitMinDur:
+		case SE_LimitInstant:
+		case SE_LimitMinLevel:
+		case SE_LimitCastTimeMin:
+		case SE_LimitCastTimeMax:
+		case SE_LimitCombatSkills:
+		case SE_LimitManaMin:
+		case SE_LimitSpellGroup:
+		case SE_LimitManaMax:
+		case SE_LimitSpellClass:
+		case SE_LimitSpellSubclass:
+		case SE_LimitClass:
+		case SE_LimitRace:
+		case SE_LimitCastingSkill:
+		case SE_LimitUseMin:
+		case SE_LimitUseType:
+		case SE_Ff_Override_NotFocusable:
+		case SE_Ff_CasterClass:
+		case SE_Ff_Same_Caster:
+		case SE_Ff_DurationMax:
+		case SE_Ff_Endurance_Max:
+		case SE_Ff_Endurance_Min:
+		case SE_Ff_ReuseTimeMin:
+		case SE_Ff_ReuseTimeMax:
+		case SE_Ff_Value_Min:
+		case SE_Ff_Value_Max:
+		case SE_Ff_FocusTimerMin:
+		case SE_FFItemClass:
+			return true;
+		default:
+			return false;
 	}
 }
 
-uint32 GetNimbusEffect(uint16 spell_id)
+int GetSpellNimbusEffect(uint16 spell_id)
 {
-	if (IsValidSpell(spell_id))
-		return (int32)spells[spell_id].nimbus_effect;
+	if (!IsValidSpell(spell_id)) {
+		return 0;
+	}
 
-	return 0;
+	return spells[spell_id].nimbus_effect;
 }
 
-int32 GetFuriousBash(uint16 spell_id)
+int GetSpellFuriousBash(uint16 spell_id)
 {
-	if (!IsValidSpell(spell_id))
+	if (!IsValidSpell(spell_id)) {
 		return 0;
+	}
 
 	bool found_effect_limit = false;
-	int32 mod = 0;
+	auto mod                = 0;
 
-	for (int i = 0; i < EFFECT_COUNT; ++i)
-		if (spells[spell_id].effect_id[i] == SE_SpellHateMod)
+	for (int i = 0; i < EFFECT_COUNT; ++i) {
+		if (spells[spell_id].effect_id[i] == SE_SpellHateMod) {
 			mod = spells[spell_id].base_value[i];
-		else if (spells[spell_id].effect_id[i] == SE_LimitEffect && spells[spell_id].base_value[i] == 999)
+		} else if (
+			spells[spell_id].effect_id[i] == SE_LimitEffect &&
+			spells[spell_id].base_value[i] == 999
+		) {
 			found_effect_limit = true;
+		}
+	}
 
-	if (found_effect_limit)
+	if (found_effect_limit) {
 		return mod;
-	else
-		return 0;
+	}
+
+	return 0;
 }
 
 bool IsShortDurationBuff(uint16 spell_id)
 {
-	if (IsValidSpell(spell_id) && spells[spell_id].short_buff_box != 0)
-		return true;
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
 
-	return false;
+	return spells[spell_id].short_buff_box != 0;
 }
 
-bool IsSpellUsableThisZoneType(uint16 spell_id, uint8 zone_type)
+bool IsSpellUsableInThisZoneType(uint16 spell_id, uint8 zone_type)
 {
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
 	//check if spell can be cast in any zone (-1 or 255), then if spell zonetype matches zone's zonetype
 	// || spells[spell_id].zonetype == 255 comparing signed 8 bit int to 255 is always false
-	if (IsValidSpell(spell_id) && (spells[spell_id].zone_type == -1 ||
-				spells[spell_id].zone_type == zone_type))
+	if (
+		spells[spell_id].zone_type == -1 ||
+		spells[spell_id].zone_type == zone_type
+	) {
 		return true;
+	}
 
 	return false;
 }
 
 const char* GetSpellName(uint16 spell_id)
 {
-    return spells[spell_id].name;
+	if (!IsValidSpell(spell_id)) {
+		return "";
```

Comment:
Would return something like "Invalid Spell" here to make it more obvious when this edge case occurs

Reply:
Makes sense to me.